### ISSUE TITLE
Issues setting up correlation between requests and pageviews

### ISSUE THREAD

**Issue by Tarig0** [Microsoft/ApplicationInsights-dotnet-server#826]
## Repro Steps
1.  created web app in visualstudio deployed to azure

##  Actual Behavior
Data is flowing in, but the operation_ids from request => page view never match
request operation ids are guid like string but the page view is a 5 character string

## Expected Behavior
Data is reported and requests map to page view when applicable

## Version Info 
SDK Version : 2.2
.NET Version :	 core 1.2						
How Application was onboarded with:VisualStudio
OS : Windows 10
Hosting Info:Azure WebApps

**Comment by MS-TimothyMothra** [comment]
Hi @Tarig0, Thanks for reporting this. 
I was able to reproduce this behavior, and I'll need to investigate this further.

**Comment by Tarig0** [comment]
Here is the relevant doc, saying this should be a feature.

https://docs.microsoft.com/en-us/azure/application-insights/application-insights-correlation

**Comment by Tarig0** [comment]
Also to note, there is correlation between Requests and dependencies, in my case an httpclient call from the server.

**Comment by MS-TimothyMothra** [comment]
@Tarig0,
I think I have a workaround for you, this worked for me in my test app.
Please try adding this script to your pages:

```
<script type="text/javascript">
    var appInsights=window.appInsights||function(a){
        function b(a){c[a]=function(){var b=arguments;c.queue.push(function(){c[a].apply(c,b)})}}var c={config:a},d=document,e=window;setTimeout(function(){var b=d.createElement("script");b.src=a.url||"https://az416426.vo.msecnd.net/scripts/a/ai.0.js",d.getElementsByTagName("script")[0].parentNode.appendChild(b)});try{c.cookie=d.cookie}catch(a){}c.queue=[];for(var f=["Event","Exception","Metric","PageView","Trace","Dependency"];f.length;)b("track"+f.pop());if(b("setAuthenticatedUserContext"),b("clearAuthenticatedUserContext"),b("startTrackEvent"),b("stopTrackEvent"),b("startTrackPage"),b("stopTrackPage"),b("flush"),!a.disableExceptionTracking){f="onerror",b("_"+f);var g=e[f];e[f]=function(a,b,d,e,h){var i=g&&g(a,b,d,e,h);return!0!==i&&c["_"+f](a,b,d,e,h),i}}return c
    }({
        instrumentationKey: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx"
    });
    
    window.appInsights=appInsights,appInsights.queue&&0===appInsights.queue.length&&appInsights.trackPageView();
</script>
```
[Source: Github/ApplicationInsights-JS](https://github.com/Microsoft/ApplicationInsights-JS/blob/master/README.md#use-js-snippet-and-initialize-dynamically-download-full-application-insights-script-from-cdn)

**Explanation:** 
[The AspNetCore dll is injecting an older version of that snippet](https://github.com/Microsoft/ApplicationInsights-aspnetcore/blob/12cb0cd96c83115eeb16d7b2711c5ddb73c9e69c/src/Microsoft.ApplicationInsights.AspNetCore/ApplicationInsightsJavaScript.cs#L17) which is causing certain client-side events to not get logged. (See screenshot where client-side dependency is missing)

![image](https://user-images.githubusercontent.com/28785781/36857173-b476be88-1d2c-11e8-9f0a-9497b0e14357.png)

**Comment by Tarig0** [comment]
interesting so the injected javascript is out of date

**Comment by Tarig0** [comment]
Should I move this issue to the AI-aspnetcore project?

**Comment by MS-TimothyMothra** [comment]
The javascript sdk version comes from https://github.com/Microsoft/ApplicationInsights-JS.

The purpose of the snippet is to download the latest version of the javascript sdk from our cdn.
While that happens, the snippet will buffer telemetry until the sdk is loaded.

To my knowledge, the snippet doesn't have a version number. I'll bring that up so maybe we can track/catch this issue in the future.

**Comment by Tarig0** [comment]
OK so I should still be seeing javascript:1.0.15 in the sdk column. I'm still seeing the short operation ids.

![image](https://user-images.githubusercontent.com/4108180/36862356-96f84d18-1d54-11e8-8868-3f4f45458ca4.png)

**Comment by MS-TimothyMothra** [comment]
I don't think we need to move this issue yet, mostly just want to spare us both the extra work 
I'll meet with the team this morning to discuss this issue.
I've already pushed a PR to the Core repo and if it's accepted I think we can close this as is.
If they ask for more work, then I'll open an Issue there to track it.

the short opIds come from javascript. the longer guid-ish ids come from core.
Try using this Analytics query to see the bigger picture of Requests, PageViews, and Dependencies:

```
(requests | union dependencies | union pageViews) 
//| where name contains "Home" or name contains "Test"
| where name !contains ".svg" and name !contains ".js" and name !contains "bootstrap/" and name !contains ".css" and name !contains ".ico"
| where timestamp >= ago(30m)
| project timestamp, itemType, sdkVersion, name, url, data, id, operation_ParentId, operation_Id
| order by timestamp
```

**Comment by Tarig0** [comment]
Not seeing any dependencies for me yet. I double checked the javascript snipet and it looks like I got it copied correctly. I'm getting the pageviews so the key didn't get botched.

I wanted to be sure to say that I appreciate the assistants so far

**Comment by MS-TimothyMothra** [comment]
@Tarig0,
Considering that we found and fixed a bug regarding this issue, I'm inclined to believe that this is no longer a server-side sdk issue. It seems to me that there's a problem with your client-side dependency telemetry not getting created. That process should be annotating your request to server-side with the necessary operation ids for correlation.

Can you please create a new issue over on https://github.com/Microsoft/ApplicationInsights-JS .
You can reference this issue for context. 

I'm also going to ask @jpiyali to follow up with you.
